content/en/guide/v10/forms.md

```diff
@@ -95,20 +97,39 @@ class MySelect extends Component {
+## Checkbox and Radio Button
+
+In HTML, browser toggles and checks checkbox and radio button. In Controlled Component, you need to implement toggle and check. Therefore, checkbox and radio button (`<input type="checkbox|radio">`) can initially cause confusion When you build Form in Controlled Component.
 
-Checkboxes and radio buttons (`<input type="checkbox|radio">`) can initially cause confusion when building controlled forms. This is because in an uncontrolled environment, we would typically allow the browser to "toggle" or "check" a checkbox or radio button for us, listening for a change event and reacting to the new value.  However, this technique does not transition well into a world view where the UI is always updated automatically in response to state and prop changes.
+Say we listen for a "change" event on a checkbox in Controlled Component, which is fired when the checkbox is checked or unchecked by the user. Form component set `checked` value in the Event object received from the checkbox to `state` in `change` event handler. This triggers rerendering component. `checked` value in `state` is reflected in the checkbox by this rerendering.
 
-> **Walk-Through:** Say we listen for a "change" event on a checkbox, which is fired when the checkbox is checked or unchecked by the user.  In our change event handler, we set a value in `state` to the new value received from the checkbox.  Doing so will trigger a re-render of our component, which will re-assign the value of the checkbox to the value from state.  This is unnecessary, because we just asked the DOM for a value but then told it to render again with whatever value we wanted.
+```jsx
+class MyForm extends Component {
+  toggle = event => {
+      const checked = event.target.checked;
+      this.setState({ checked });
+  };
 
-So, instead of listening for a `input` event we should listen for a `click` event, which is fired any time the user clicks on the checkbox _or an associated `<label>`_.  Checkboxes just toggle between Boolean `true` and `false`, so clicking the checkbox or the label, we'll just invert whatever value we have in state, triggering a re-render, setting the checkbox's displayed value to the one we want.
+  render(_, { checked }) {
+    return (
+      <label>
+        <input
+          type="checkbox"
+          checked={checked}
+          onClick={this.toggle}
+        />
+      </label>
+    );
+  }
+}
+```
 
-### Checkbox Example
+But, it is unnecessary to use `checked` value in Event object, since the component has `checked` value at before changing in `state`. So, instead of listening for a `change` event we should listen for a `click` event, which is fired any time the user clicks on the checkbox _or an associated `<label>`_. Checkboxes just toggle between Boolean `true` and `false`, so clicking the checkbox or the label, we'll just invert whatever value we have in state, triggering a re-render, setting the checkbox's displayed value to the one we want.
```

Comment:
```suggestion
When managing the state of checkboxes or radio buttons, it's important to understand the two different mechanisms for observing value changes. The `click` event is triggered on an input when its own value changes, such as when a radio button is clicked, or activated via the keyboard or its associated `<label>`. The `change` event is triggered on all inputs in a form group when the value of that _group_ is changed.

It's most common to use the `click` event for checkboxes and radio buttons, since it doesn't rely on named form groups which are a type of DOM state. When handling `click` events, the input element's boolean `checked` property indicates its current value.

In the example below, clicking on the checkbox or its label should toggle between Boolean `true` and `false` values. To achieve this as a controlled input, we can store the initial value in state and simply invert it in response to clicks. This will trigger a re-render, setting the checkbox's displayed value to the value from state.
```